elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchProjectionConstants.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch;
+
+/**
+ * Projection constants specific to the Elasticsearch backend.
+ *
```

Comment:
Maybe we should mention in the comment that they also have to be declared in `SUPPORTED_PROJECTION_CONSTANTS`.

That's where an `enum` would have been handy. Any reason why we dont switch to an enum implementing an interface (true question, interested in the answer!)? I think we could do this while keeping the backwards compatibility. It wouldn't buy us any more type safety though.